java-checks-test-sources/src/main/java/checks/UnreachableCatchCheck.java

```diff
@@ -13,26 +13,26 @@ void unreachable(boolean cond) {
       throwExtendsCustomException();
     } catch (ExtendsCustomException e) {
       // ...
-    } catch (CustomException e) { // Noncompliant [[sc=7;ec=12;secondary=14]] {{Remove or refactor this catch clause because it is unreachable as hidden by previous catch blocks.}}
+    } catch (CustomException e) { // Noncompliant [[sc=7;ec=12;secondary=14]] {{Remove or refactor this catch clause because it is unreachable, hidden by previous catch block(s).}}
       // ...
     }
 
     try {
       throw new ExtendsCustomException();
     } catch (ExtendsCustomException e) {
       // ...
-    } catch (CustomException|IllegalStateException e) { // Noncompliant [[sc=14;ec=29]] {{Remove this type because it is unreachable as hidden by previous catch blocks.}}
+    } catch (CustomException|IllegalStateException e) { // Noncompliant [[sc=14;ec=29;secondary=22]] {{Remove this type because it is unreachable, hidden by previous catch block(s).}}
       // ...
     }
 
     try {
       if (true) throw new ExtendsExtendsCustomException();
-      else throw new ExtendsOtherExtendsCustomException();
+      throw new ExtendsOtherExtendsCustomException();
     } catch (ExtendsExtendsCustomException e) {
       // ...
     } catch (ExtendsOtherExtendsCustomException e) {
-      // ...
-    } catch (ExtendsCustomException|OtherExtendsCustomException e) { // Noncompliant [[sc=7;ec=12;secondary=31,33]] {{Remove or refactor this catch clause because it is unreachable as hidden by previous catch blocks.}}
+      // Noncompliant@+1
+    } catch (ExtendsCustomException | OtherExtendsCustomException e) { // Noncompliant
```

Comment:
Here we still tell the user to remove all types from the catch clause rather than telling them to just remove the catch clause.